daft/io/_range.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+from daft import DataType
+from daft.io._generator import GeneratorScanOperator
+from daft.logical.schema import Schema
+from daft.table.table import Table
+
+
+def _range_generators(start: int, end: int, step: int) -> Iterator[Callable[[], Iterator[Table]]]:
+    def generator_for_value(value: int) -> Callable[[], Iterator[Table]]:
+        def generator() -> Iterator[Table]:
```

Comment:
this is generating 1 row per table which will be extremely slow. Instead you should be calculating what range should go in each partition and then call something like `Series.arange` to generate the series in rust. Each partition can just be 1 table.

Reply:
kinda confused what you are thinking interop will be like in this case. Are you thinking 

- Series.arange is in rust
- table gets sent to python (Iterator of Tables)
- get python RangeScanOperator
- then get rust binding of RangeScanOperator (`let scan_operator_handle = ScanOperatorHandle::from_python_scan_operator(range, py)?;`)

Reply:
@samster25 I added partitions in 

f58d2ae3874783450dfbb08b43af56df504d11f3

also added #3334 so we can switch over to pure `arange` approach

Reply:
resolving this as no response